daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -48,6 +50,12 @@ final class Engine {
   private[this] val _commandTranslation: CommandPreprocessor = new CommandPreprocessor(
     _compiledPackages)
 
+  // We want to make the submission different from the ledgerTime,
+  // then we pick a random time +/- 5ms around ledger time
```

Comment:
@andreaslochbihler-da, what should be the delta ?

Reply:
+/-5 or 10 microseconds are both fine.

Two suggestions for the implementation:
1. Keep your own random instance in this class that users can configure. This will help to make tests deterministic (they can provide the seed).
2. I don't see any need for using a normal distribution here; the uniform distribution should be just as good and that requires less computation than the Gaussian.

Reply:
Thanks Andrea.
I implemented your suggestions.